src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -52,18 +52,31 @@ public Connection(Uri url, ILoggerFactory loggerFactory)
 
         public Task StartAsync(ITransport transport, HttpClient httpClient)
         {
-            _startTask = StartAsyncInternal(transport, httpClient);
-            return _startTask;
-        }
-
-        private async Task StartAsyncInternal(ITransport transport, HttpClient httpClient)
-        {
             if (Interlocked.CompareExchange(ref _connectionState, ConnectionState.Connecting, ConnectionState.Initial)
                 != ConnectionState.Initial)
             {
-                throw new InvalidOperationException("Cannot start a connection that is not in the Initial state.");
+                return Task.FromException(
+                    new InvalidOperationException("Cannot start a connection that is not in the Initial state."));
             }
 
+            StartAsyncInternal(transport, httpClient)
+                .ContinueWith(t =>
+                {
+                    if (t.IsFaulted)
```

Comment:
StartAsyncInernal never returns a cancelled task. Do you think we should add this proactively?